packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -166,11 +166,9 @@ const _ActionMenuRoot = memo(
 
     const handleItemSelection = useCallback(
       (newValue: string) => {
-        if (newValue) {
-          handleOnSelect(newValue)
-          toggleMenu()
-          anchorElementRef.current?.focus()
-        }
+        handleOnSelect(newValue)
+        toggleMenu()
+        anchorElementRef.current?.focus()
```

Comment:
Do we need to remove the guard entirely if we want to only allow "" values to pass through? Could we continue to guard against other falsy values?

Something like:

```suggestion
        if (newValue || newValue === "") {
          handleOnSelect(newValue)
          toggleMenu()
          anchorElementRef.current?.focus()
        }
```

Reply:
It's a good spot, and I did have that at first, however I noticed that it was a bit redundant as `newValue`'s type is a (non-optional) string, so the only scenario in which it can ever be falsy is when it equals `""`. I didn't want to add redundant checks in that TypeScript already covered for us, so I ended up with this.

Reply:
> Curious what the use-cases are for allowing all falsy values through otherwise.

In reality I'm not sure there's ever a situation where a falsy value is the only way to accomplish something, but the current behaviour does feel a bit counterintuitive.

More context on the specific use-case that prompted this here https://github.com/primer/brand/issues/906

Reply:
Thanks, I've just looked and the original issue and wondering if this is even feasible.

We can't allow falsy values because the selected text is needed for the button label. This is why the guard is likely there to begin with. 

https://github.com/user-attachments/assets/34e67ec3-7eef-46d8-b20c-ff07ca8b1876

Reply:
From what I can tell, the `ActionMenu` doesn't manage the Button label. The label gets passed in as a child to `<ActionMenu.Button>` so it's up to the implementor to define. In many of our Storybook and Docs examples we have added some state to the rendered example to implement the behaviour shown in your video above, but this functionality is not actually a part of the component itself.

Assuming `ActionMenu` is intended to function in the same way as an HTML `<select>` element, then we should treat a value of `""` in the same way as any other string since `<select>` doesn't discriminate between `""` and any other string. In the case of `<select>`, using an empty string as a value doesn't seem to be an uncommon pattern.

Reply:
Ah yes, just saw that thanks. That's fine then. I guess it's okay to make this change, but I still can't really see the use-case for allowing null or undefined values through like this. I guess it means that the user has to mitigate those scenario's where data is from unreliable sources themselves?

Reply:
Yea potentially.

Equally we could approach the question from the other side and ask if there's a good reason that we _should_ impose restrictions on the strings that are passed. Maybe there's no situation where passing an empty value is the _only_ way to make something work, but it seems reasonable that we offer that flexibility to consumers in case that's the approach they want to take.

Just to confirm too; null or undefined values will be flagged by TypeScript, the only behaviour that is modified by this PR is specifically the `value=""` scenario.

Reply:
I see 🤔 PB does need to work in non-TS contexts though that don't have type inference. Want to make sure we're not making assumptions about that. But regardless, I agree with:

> it seems reasonable that we offer that flexibility to consumers in case that's the approach they want to take

👍